cheetah/particles/particle_beam.py

```diff
@@ -883,6 +898,351 @@ def to_xyz_pxpypz(self) -> torch.Tensor:
 
         return xp_coords
 
+    def plot_1d_distribution(
+        self,
+        dimension: Literal["x", "px", "y", "py", "tau", "p"],
+        bins: int = 100,
+        bin_range: Optional[Tuple[float]] = None,
+        smoothing: float = 0.0,
+        plot_kws: Optional[dict] = None,
+        ax: Optional[plt.Axes] = None,
+    ) -> plt.Axes:
+        """
+        Plot a 1D histogram of the given dimension of the particle distribution.
+
+        :param dimension: Name of the dimension to plot. Should be one of
+            `('x', 'px', 'y', 'py', 'tau', 'p')`.
+        :param bins: Number of bins to use for the histogram.
+        :param bin_range: Range of the bins to use for the histogram.
+        :param smoothing: Standard deviation of the Gaussian kernel used to smooth the
+            histogram.
+        :param plot_kws: Additional keyword arguments to be passed to `plot` function of
+            matplotlib used to plot the histogram data.
+        :param ax: Matplotlib axes object to use for plotting.
+        :return: Matplotlib axes object with the plot.
+        """
+        if ax is None:
+            _, ax = plt.subplots()
+
+        x_array = getattr(self, dimension).cpu().detach().numpy()
+        histogram, edges = np.histogram(x_array, bins=bins, range=bin_range)
+        centers = (edges[:-1] + edges[1:]) / 2
+
+        if smoothing:
+            histogram = gaussian_filter(histogram, smoothing)
+
+        ax.plot(
+            centers,
+            histogram / histogram.max(),
+            **{"color": "black"} | (plot_kws or {}),
+        )
+        ax.set_xlabel(f"{self.PRETTY_DIMENSION_LABELS[dimension]}")
+
+        # Handle units
+        if dimension in ("x", "y", "tau"):
+            base_unit = "m"
+        elif dimension in ("px", "py"):
+            base_unit = "rad"
+        elif dimension == "p":
+            base_unit = "%"
```

Comment:
I think the proper way to say is $px, py, p$ are all dimensionless. 

- px,py are not exactly the geometrical slope, so using $rad$ might be misleading
- p is just a ratio, if expressed in % it still needs to be multiplied by 100

Reply:
Ah ... I didn't realise that about $px$ and $py$. With $p$, the tick formatter does the multiplication by 100 ... I'm not sure if the variable name `base_unit` is a bit misleading here in that case (it was just `unit` before).

So you would propose to do all three in % then?

Reply:
I would suggest to put them as dimensionless (no unit), if that's possible.

Reply:
So kind of how I changed it now?